rfc/0002/README.md

```diff
@@ -0,0 +1,169 @@
+    3. the NNF storage requirements described above. 
+3. The ADMINISTRATOR creates a corresponding _NNF Container Profile_ custom kubernetes resource
+4. The USER who desires to use the application works with the AUTHOR and the related Container Profile to understand the storage requirements.
+5. The USER submits a WLM job with the #DW container fields populated
+6. WLM runs the job and drives the job through the following stages...
+    1. Proposal: RABBIT validates the #DW container directive by comparing the supplied values to what is listed in the Container Profile. If the USER fails to meet the requirements, the job fails. 
+    2. Pre-run: RABBIT software will:
+        1. create a config map reflecting the storage requirements and any runtime parameters
+        2. duplicate the pod specification from the Container Profile and patches the necessary Volumes and the config map. The spec is used as the basis for starting the necessary pods and containers.
+    3. The containerized application executes. The expected mounts are available per the requirements and celebration occurs.
+
+Example
+-------
+
+Say I authored a simple application, `foo`, that requires Rabbit local GFS2 storage and a persistent Lustre storage volume. As the author, my program is coded to expect the GFS2 volume is mounted at `/foo/local` and the Lustre volume is mounted at `/foo/persistent`
+
+Working with an administrator, my applications storage requirements and pod specification are placed in an NNF Container Profile `foo`
+
+```yaml
+kind: NnfContainerProfile
+apiVersion: v1alpha1
+metadata:
+    name: foo
+    namespace: default
+spec:
+    storages:
+    - name: $JOB_DW_foo-local-storage
+      type: gfs2
+    - name: $PERSISTENT_DW_foo-persistent-storage
+      type: lustre
+    podSpec:
+        containers:
+        - name: foo
+          image: foo:latest
+          command:
+          - /foo
+          volumeMounts:
+          - name: foo-local-storage
+            mountPath: /foo/local
+          - name: foo-persistent-storage
+            mountPath: /foo/persistent
+```
+
+Say Peter wants to use `foo` as part of his job specification. Peter would submit the job with the below directives
+
+```
+#DW jobdw name=my-gfs2 type=gfs2 capacity=1TB
+
+#DW persistentdw name=some-lustre
+
+#DW container name=my-foo profile=foo                 \
+    $JOB_DW_foo-local-storage=my-gfs2                 \
+    $PERSISTENT_DW_foo-persistent-storage=some-lustre
+```
+
+Peter submits the job to the WLM. WLM guides the job through the workflow states:
+1. Proposal: Rabbit software verifies the #DW directives. For the container directive, the requirements of Container Profile `foo`, that is `foo-local-storage` and `foo-persistent-storage`, are provided and correct.
+2. Setup: Since there is a jobdw `my-gfs2`, Rabbit software provisions this storage.
+3. Pre-Run:
+    1. Rabbit software generates a config map that corresponds to the storage requirements and runtime parameters.
```

Comment:
```suggestion
    1. Rabbit software generates a `ConfigMap` that corresponds to the storage requirements and runtime parameters.
```

Reply:
In the proposal and example I tend to use words rather than the kubernetes resource Kind value